### PR TITLE
Add ADC support for the NXP KE1xF SoC and TWR-KE18F development board

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This patch series adds ADC support for the NXP Kinetis KE1xF SoC series and the NXP TWR-KE18F development board:
- drivers: adc: mcux_adc12: add driver for the NXP ADC12 module
- soc: nxp: ke1xf: add ADC12 support
- boards: arm: twr_ke18f: add ADC support
- tests: drivers: adc_api: add support for the twr_ke18f board

The add_api test case still fails on the target due to having userspace enabled. This is being handled in #16399.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc change LGTM

File: soc/arm/nxp_kinetis/ke1xf/dts_fixup.h [hunk 1]
```diff
@@ -74,3 +74,30 @@
...
 #define DT_I2C_MCUX_LPI2C_1_CLOCK_SUBSYS	DT_NXP_IMX_LPI2C_40067000_CLOCK_NAME
+
+#define DT_ADC_0_BASE_ADDRESS			DT_NXP_KINETIS_ADC12_4003B000_BASE_ADDRESS
+#define DT_ADC_0_IRQ				DT_NXP_KINETIS_ADC12_4003B000_IRQ_0
+#define DT_ADC_0_IRQ_PRI			DT_NXP_KINETIS_ADC12_4003B000_IRQ_0_PRIORITY
+#define DT_ADC_0_NAME				DT_NXP_KINETIS_ADC12_4003B000_LABEL
+#define DT_ADC_0_CLK_SOURCE			DT_NXP_KINETIS_ADC12_4003B000_CLK_SOURCE
+#define DT_ADC_0_CLK_DIVIDER			DT_NXP_KINETIS_ADC12_4003B000_CLK_DIVIDER
+#define DT_ADC_0_ALTERNATE_VOLTAGE_REFERENCE	DT_NXP_KINETIS_ADC12_4003B000_ALTERNATE_VOLTAGE_REFERENCE
+#define DT_ADC_0_SAMPLE_TIME			DT_NXP_KINETIS_ADC12_4003B000_SAMPLE_TIME
```
Comment (MaureenHelm) [inline soc/arm/nxp_kinetis/ke1xf/dts_fixup.h]: You can skip all the soc dts fixups if you change the adc driver to use `DT_NXP_KINETIS_ADC12_ADC_0_BASE_ADDRESS` and friends.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: soc/arm/nxp_kinetis/ke1xf/dts_fixup.h (hunk 1 above)
Comment (henrikbrixandersen) [reply soc/arm/nxp_kinetis/ke1xf/dts_fixup.h]: Sure. I was just modeling it like most of the other MCUX driver shims and added an extra layer of indirection to not tie it to the Kinetis series. 

If you prefer not having this extra indirection, I’ll get rid of it.

Comment (henrikbrixandersen) [reply soc/arm/nxp_kinetis/ke1xf/dts_fixup.h]: @MaureenHelm Fixed.

**Review by MaureenHelm — APPROVED** [review]

File: soc/arm/nxp_kinetis/ke1xf/dts_fixup.h (hunk 1 above)
Comment (MaureenHelm) [reply soc/arm/nxp_kinetis/ke1xf/dts_fixup.h]: We probably need to go back and clean up the other mcux driver shims now that we can use aliases this way. I think it's a lot cleaner not to have all the dts fixups.

**Pull request merged**